.travis.yml

```diff
@@ -6,34 +6,29 @@ jdk:
 cache:
   directories:
    - $HOME/.m2
+   - $HOME/google-cloud-sdk
+env:
+  global:
+    - CLOUDSDK_CORE_DISABLE_USAGE_REPORTING=true
+    - CLOUDSDK_CORE_DISABLE_PROMPTS=true
 install:
   # https://github.com/travis-ci/travis-ci/issues/8408
   - unset _JAVA_OPTIONS
-  # download Cloud SDK
-  - wget https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-sdk-200.0.0-linux-x86_64.tar.gz
-  - tar -xzf google-cloud-sdk-200.0.0-linux-x86_64.tar.gz -C /home/travis
-  # update all Cloud SDK components
-  - /home/travis/google-cloud-sdk/bin/gcloud components update --quiet
+
+  # travis auto creates cache directories, so remove it
+  # run install script
+  - if [ ! -d $HOME/google-cloud-sdk/bin ]; then
```

Comment:
I think it's that if `google-cloud-sdk` wasn't cached (for example, the cache was trashed) then Travis will create the base directory as specified in the `cache` section, but won't create any of the contents.  So if `google-cloud-sdk/bin` doesn't exist, then the contents weren't found in the cache.

Reply:
Aha, that explains all. Thanks.

Reply:
I think a better comment would be "travis auto creates cache directories even if nothing is cached, so check a subdirectory" or such.

Reply:
ah, the check is for `$HOME/google-cloud-sdk`**`/bin`** which is the indicator that a cloud sdk has been installed?

Then we delete the `$HOME/google-cloud-sdk` because the cloud sdk wants to install in a new directory.

Reply:
The comments are also weird, because I don't think you can embed comments in the middle of the bash command in the yaml

Reply:
YAML comments can appear in collections ([spec](https://yaml.org/spec/current.html#id2509980))

Reply:
hrmm.. maybe I done something else dumb. Gonna submit this and test that out.